File: tutorial/examples/faqs/last_clicked_button.py
import json

import dash
import dash_html_components as html
from dash.dependencies import Input, Output

app = dash.Dash(__name__)

app.layout = html.Div([
    html.Button('Button 1', id='btn-1'),
    html.Button('Button 2', id='btn-2'),
    html.Button('Button 3', id='btn-3'),
    html.Div(id='container')
])


@app.callback(Output('container', 'children'),
              [Input('btn-1', 'n_clicks'),
               Input('btn-2', 'n_clicks'),
               Input('btn-3', 'n_clicks')])
def display(btn1, btn2, btn3):
    ctx = dash.callback_context

    if not ctx.triggered:
        msg = 'None of the buttons have been clicked yet'
    else:
        button_num = ctx.triggered[0]['prop_id'].split('.')[0].split('-')[1]
        msg = 'Button {} was most recently clicked'.format(button_num)

    ctx_msg = json.dumps({
        'states': ctx.states,
        'triggered': ctx.triggered,
        'inputs': ctx.inputs
    }, indent=2)

    return html.Div([
        html.Div('btn1: {}'.format(btn1)),
        html.Div('btn2: {}'.format(btn2)),
        html.Div('btn3: {}'.format(btn3)),
        html.Div(msg),
        html.Pre(ctx_msg)
    ])


if __name__ == '__main__':
    app.run_server(debug=True)

File: dashr/chapters/faq-gotchas/examples/last_clicked_button.R
library(dashR)
library(dashCoreComponents)
library(dashHtmlComponents)

app <- Dash$new()

app$layout(html.Div(list(
  htmlButton('Button 1', id='btn-1', n_clicks=0, children='Submit'),
  htmlButton('Button 2', id='btn-2', n_clicks=0, children='Submit'),
  htmlButton('Button 3', id='btn-3', n_clicks=0, children='Submit'),
  htmlDiv(id='container')
  
)))

app$callback(output('container', 'children'),
             list(input('btn-1', 'n_clicks'),
                  input('btn-2', 'value'),
                  input('btn-3', 'value')),
    function(n_clicks, input1, input2) {
    return(htmlDiv(list(
      htmlTable(list(
        htmlTr(list(htmlTh('Button 1'),
                 htmlTh('Button 2'),
                 htmlTh('Button 3'),
                 htmlTh('Most Recent Click'))),
        htmlTr(list(htmlTd(btn1),
                 htmlTd(btn2),
                 htmlTd(btn3),
                 htmlTd(button_id)))
        ))
      )))
    })
#app$run_heroku()
